manager/allocator/networkallocator/portallocator.go

```diff
@@ -91,13 +91,17 @@ func reconcilePortConfigs(s *api.Service) []*api.PortConfig {
 		return s.Spec.Endpoint.Ports
 	}
 
-	allocatedPorts := make(map[api.PortConfig]*api.PortConfig)
+	allocatedPorts := make(map[api.PortConfig]map[uint32]*api.PortConfig)
 	for _, portState := range s.Endpoint.Ports {
 		if portState.PublishMode != api.PublishModeIngress {
 			continue
 		}
 
-		allocatedPorts[getPortConfigKey(portState)] = portState
+		portKey := getPortConfigKey(portState)
+		if _, ok := allocatedPorts[portKey]; !ok {
+			allocatedPorts[portKey] = make(map[uint32]*api.PortConfig)
```

Comment:
Wondering; if `getPortConfigKey()` is intended to return the unique identifier for a port; should `PublishedPort` be just added to what `getPortConfigKey()` returns, instead of adding a `map` here?